docs/pywbemcligeneraloptions.rst

```diff
@@ -61,166 +70,470 @@ command, general the options redefinition's are not retained between command
+
+For more information on authentication types, see:
+:https://pywbem.readthedocs.io/en/stable/client/security.html#authentication-types
+
+.. _`--keyfile general option`:
+
+----keyfile general option
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The value of the ``--keyfile`` general option is a file path of the
+the client private key file containing the private key belonging to the public
+key that is part of the X.509 certificate. See ``--certfile`` for more
+information. Not required if the private key is part of the file defined in
+``--certfile``. Not allowed if ``--certfile`` option not provided. Default: No
+client key file. Client private key should then be part of the file defined by
+``--certfile``.
+
+The pywbemcli command line options are as follows (The following complements
+the exact help text output shown in section :ref:`pywbemcli Help Command
+Details`)
+
+
+.. _`--ca-certs general option`:
+
+----ca-certs general option
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The value of the ``--ca-certs`` a file or directory containing certificates
+that will be matched against acertificate received from WBEM server. Setting
+the :ref:`--no-verify general option` bypasses client verification of the
+WBEM server certificate. The default is OS dependent and is a set of system
+directories where certficates are expected to be stored for the client OS.
+
+
+.. _`--use-pull general option`:
+
+--use_pull general option
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The value of the ``--use-pull/-u`` general option is one of the following
+strings: [``yes``|``no``|``either``] that determines whether the pull
+operations are used for ``EnumerateInstances``, ``AssociatorInstances``,
+``ReferenceInstances``, and ``ExecQuery`` operations See :ref:`Pywbemcli and
+the DMTF pull operations` for more information on pull operations. The value
+choices are as follows:
+
+* ``yes`` - pull operations will be used and if the server does not
+  support pull, the request will fail.
+* ``no`` - forces pywbemcli to try only the traditional non-pull operations.
+* ``either`` - (default) pywbem tries both; first pull operations and then
+  :term:`traditional operations` .
+
+.. _`--pull-max-cnt general option`:
+
+--pull-max-cnt general option
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The value of the ``--pull-max-cnt`` general option is an integer passed to the
+WBEM server with the open and pull operation requests. This integer,
+``MaxObjectCount`` tells the server the maximum number of objects
+to be returned for each pull request if pull operations are used. This must
+be  a positive non-zero integer. The default is 1000. See :ref:`Pywbemcli and the
+DMTF pull operations` for more information on pull operations.
+
+.. _`--mock-server general option`:
+
+--mock-server general option
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The value of the ``--mock-server/-m`` general option is a file path that
+defines a mock server that can be used to define a mock WBEM server in the
```

Comment:
"defines a mock server that can be used to define a mock WBEM server" could be somewhat more condensed.

Reply:
Rewrote.